tensorflow/compiler/xla/service/buffer_assignment.cc

```diff
@@ -1330,12 +1330,23 @@ Status BufferAssigner::AssignBuffersWithSequentialOrdering(
   auto get_heap_algorithm = [&](int64 alignment) {
     auto algorithms = absl::make_unique<
         std::vector<std::unique_ptr<HeapAlgorithm<HloValue>>>>();
-    algorithms->push_back(
-        absl::make_unique<GlobalDecreasingSizeBestFitHeap<HloValue>>(
-            alignment, GlobalDecreasingSizeBestFitHeap<HloValue>::kSpatial));
-    algorithms->push_back(
-        absl::make_unique<GlobalDecreasingSizeBestFitHeap<HloValue>>(
-            alignment, GlobalDecreasingSizeBestFitHeap<HloValue>::kTemporal));
+    if (assignment->multiheap_size_constraint_per_heap() == -1) {
+      algorithms->push_back(
+          absl::make_unique<GlobalDecreasingSizeBestFitHeap<HloValue>>(
+              alignment, GlobalDecreasingSizeBestFitHeap<HloValue>::kSpatial));
+      algorithms->push_back(
+          absl::make_unique<GlobalDecreasingSizeBestFitHeap<HloValue>>(
+              alignment, GlobalDecreasingSizeBestFitHeap<HloValue>::kTemporal));
+    } else {
+      algorithms->push_back(
```

Comment:
Instead of switching between the two algorithms, would it make more sense to instead always use the `Constrained***` algorithm, and parameterize it by the constraint size (infinitely/very large in unconstrained case)?

Reply:
It makes sense. Let me revise accordingly.